### ISSUE TITLE
issue with dependencies in node_modules

### ISSUE THREAD

**Issue by mathib** [ali1k/ld-r#31]
## Expected Behavior
An LD-R app that can run in dev mode and can be build on windows.

## Actual Behavior
The webpack-dev-server respectively webpack module could not be found after plain install of the default repository:
```
> linked-data-reactor@1.1.11 dev C:\Users\USER-NAME\Documents\LD-R\ld-r3
> node ./webpack/dev-server.js

module.js:557
    throw err;
    ^

Error: Cannot find module 'webpack-dev-server'
    at Function.Module._resolveFilename (module.js:555:15)
    at Function.Module._load (module.js:482:25)
    at Module.require (module.js:604:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (C:\Users\USER-NAME\Documents\LD-R\ld-r3\webpack\dev-server.js:1:86)
    at Module._compile (module.js:660:30)
    at Object.Module._extensions..js (module.js:671:10)
    at Module.load (module.js:573:32)
    at tryModuleLoad (module.js:513:12)
    at Function.Module._load (module.js:505:3)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! linked-data-reactor@1.1.11 dev: `node ./webpack/dev-server.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the linked-data-reactor@1.1.11 dev script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\USER-NAME\AppData\Roaming\npm-cache\_logs\2018-02-10T20_07_26_688Z-debug.log
```
If I check the packages in the local node_modules, I noticed that only the half of the packages are there compared to an older LD-R v1.1.9 app that was working (only in dev mode on windows, it could not be build). After manually copying the packages, I could get the LD-R v1.1.11 app running in dev mode and I could build it on Windows.

## Steps to Reproduce the Problem

  1. Install the default LD-R web application from this repo following the Quick Start guide on a windows machine
  2. Run the app in dev mode (`npm run dev`) or build it (`npm run build:windows`)

## Specifications

  - Version: LD-R v1.1.11
  - Platform: Windows 10 (Node v9.4.0, npm v5.6.0, webpack v3.10.0 (globally installed))

**Comment by mathib** [comment]
I tried to install using the ìnstall.bat`. I also tried to install it using the manual way with `npm install`, but also without result. This was the output of `npm install`:
```
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN request-promise@4.2.2 requires a peer of request@^2.34 but none is installed. You must install peer dependencies yourself.
npm WARN request-promise-core@1.1.1 requires a peer of request@^2.34 but none is installed. You must install peer dependencies yourself.

added 244 packages in 20.795s
```

**Comment by mathib** [comment]
When I try to install webpack and webpack-dev-server locally, I get the following:
```
C:\Users\USER-NAME\Documents\LD-R\ld-r5>npm install --no-optional --save-dev webpack webpack-dev-server
npm WARN request-promise@4.2.2 requires a peer of request@^2.34 but none is installed. You must install peer dependencies yourself.
npm WARN request-promise-core@1.1.1 requires a peer of request@^2.34 but none is installed. You must install peer dependencies yourself.

up to date in 13.284s
```
But in the node_modules folder, there is still no webpack or webpack-dev-server

**Comment by ali1k** [comment]
it seems to be a general npm issue on Windows rather than LD-R issue! Need some time to reproduce it on a Windows machine.

**Comment by mathib** [comment]
The strange thing is, that with LD-R v1.1.9, I was able to run the app in dev mode (webpack and webpack-dev-server were installed correctly then)...

Anyway, I did a workaround: by manually copying the node_modules folder from the older LD-R project I had (v1.1.9) I could get it going for now. But this can cause issues for new windows users of course.

**Comment by ali1k** [comment]
I checked this issue again on a Windows 10 machine and all the node_modules were successfully added. The only issue I noticed was setting the env variable for the build mode on windows which sounds problematic. As I see it can set the env car but cannot retrieve it correctly.

**Comment by mathib** [comment]
It's still not solved for my case... This is a list of the packages that are installed in the node_modules folder:
 [f.txt](https://github.com/ali1k/ld-r/files/1728155/f.txt)

**Comment by mathib** [comment]
I wanted to give an update on this:
* If I run LD-R on my Mac, it don't give any problems for dev and build mode
* If I want to run a new LD-R clone on my windows 10, it keeps failing on installing the required packages when I run the install script. For some reason, it worked to run LD-R in dev mode earlier on (I think somewhere in January of this year). Any idea on how to proceed on Windows?

P.S. I've found a temporal workaround by just copying the node_modules folder of the older working LD-R instance to the new LD-R. If I did this, I can run LD-R in dev and product mode without any problem (I need the `npm run build:windows` command if I want to build it). This clumsy method will probably give some issues regarding to dependencies in the future, I guess?

**Comment by mathib** [comment]
@ali1k  I have the feeling that it has something to do with my Windows being not able to find the globally installed webpack. If I check the version from the command line, it finds it without problems, but when I run clone LD-R and let it install its packages, it does not add the webpack related packages.

Maybe I can try to install webpack locally, but I don't know if this is possible/recommended for LD-R. If I run `npm install webpack --save-dev` (without having a webpack installed globally) it tells me that webpack is already up to date:

> C:\Users\USER-NAME\Documents\LD-R\ld-r12>npm install --save-dev webpack
> npm notice created a lockfile as package-lock.json. You should commit this file.
> up to date in 13.041s

Any ideas on how to proceed regarding a local webpack installation? If it mathers, I use a node version manager called [nvm-windows](https://github.com/coreybutler/nvm-windows) v1.1.6

**Comment by ali1k** [comment]
@mathib that's weird! maybe it is a problem with the path of Webpack in the Windows.
The best way to resolve the issue is to have a call where we can try different options.

**Comment by mathib** [comment]
For anyone interested in my issue: it was related to a NODE_ENV system variable set to 'production'. This made it impossible to install with `npm install` the devDependencies such as Webpack and others. I've used this [source](https://stackoverflow.com/questions/34700610/npm-install-wont-install-devdependencies).

I've changed the NODE_ENV to 'development'. To be certain, I've also checked the `npm config get production` and made sure it was set to false (`npm config set -g production false`)

@ali1k  Thanks for the assistence! Issue can be closed :)
